cpp/include/raft/distance/detail/cosine.cuh

```diff
@@ -248,10 +247,13 @@ void cosineAlgo1(Index_ m,
   InType* row_vec = workspace;
   if (pA != pB) {
     row_vec += m;
-    raft::linalg::rowNorm(col_vec, pA, k, m, raft::linalg::L2Norm, isRowMajor, stream, norm_op);
-    raft::linalg::rowNorm(row_vec, pB, k, n, raft::linalg::L2Norm, isRowMajor, stream, norm_op);
+    raft::linalg::rowNorm(
```

Comment:
Well, this code doesn't compile at all if `AccType != InType`.

```c++
AccType* workspace,
...
InType* col_vec = workspace;
```

Since it affects the other distance functions and it's not related to this PR, I'll open a separate issue rather than adding comments.